perun-core/src/main/java/cz/metacentrum/perun/core/api/ConsentsManager.java

```diff
@@ -1,10 +1,62 @@
 package cz.metacentrum.perun.core.api;
 
+import cz.metacentrum.perun.core.api.exceptions.ConsentHubNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.FacilityNotExistsException;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+import cz.metacentrum.perun.core.api.exceptions.PrivilegeException;
+
+import java.util.List;
+
 /**
  * Consents entry logic.
  *
  * @author Radoslav Čerhák <r.cerhak@gmail.com>
  */
 public interface ConsentsManager {
 
+	/**
+	 * Get list of all Consent Hubs
+	 *
+	 * @param sess perun session
+	 * @return list of Consent Hubs
+	 * @throws PrivilegeException
+	 * @throws InternalErrorException
+	 */
+	List<ConsentHub> getAllConsentHubs(PerunSession sess) throws PrivilegeException;
+
+	/**
+	 * Finds existing Consent Hub by id.
+	 *
+	 * @param sess perun session
+	 * @param id id of the Consent Hub you are looking for
+	 * @return found Consent Hub
+	 * @throws ConsentHubNotExistsException
+	 * @throws InternalErrorException
+	 */
+	ConsentHub getConsentHubById(PerunSession sess, int id) throws ConsentHubNotExistsException, PrivilegeException;
+
+	/**
+	 * Finds existing Consent Hub by name.
+	 *
+	 * @param sess perun session
+	 * @param name name of the Consent Hub you are looking for
+	 * @return found Consent Hub
+	 * @throws ConsentHubNotExistsException
+	 * @throws InternalErrorException
+	 */
+	ConsentHub getConsentHubByName(PerunSession sess, String name) throws ConsentHubNotExistsException, PrivilegeException;
+
+	/**
+	 * Finds existing Consent Hub by facility.
+	 *
+	 * @param sess perun session
+	 * @param facilityId id of facility for which consent hub is searched
+	 * @return found Consent Hub
+	 * @throws ConsentHubNotExistsException
+	 * @throws InternalErrorException
+	 * @throws PrivilegeException
+	 * @throws FacilityNotExistsException
+	 */
+	ConsentHub getConsentHubByFacility(PerunSession sess, int facilityId) throws ConsentHubNotExistsException, PrivilegeException, FacilityNotExistsException;
```

Comment:
The task states that in the **future** 1 facility will be a part of max 1 consent hub, are we implementing it now? If not, then this could potentially return multiple CHs, no? I also agree that this should be implemented now, but just make sure that everybody is aware. (e.g. https://github.com/CESNET/perun/pull/3528 doesn't deal with this), maybe update associated tasks?

Reply:
Should be solved in the PR you mentioned. We want 1 hub/facility now.